src/mbgl/gl/gl_functions.cpp

```diff
@@ -0,0 +1,333 @@
+void (*GLFunctions::glTexParameterf)(GLenum, GLenum, GLfloat);
+void (*GLFunctions::glTexParameterfv)(GLenum, GLenum, const GLfloat *);
+void (*GLFunctions::glTexParameteri)(GLenum, GLenum, GLint);
+void (*GLFunctions::glTexParameteriv)(GLenum, GLenum, const GLint *);
+void (*GLFunctions::glTexSubImage2D)(GLenum, GLint, GLint, GLint, GLsizei, GLsizei, GLenum, GLenum, const void *);
+void (*GLFunctions::glUniform1f)(GLint, GLfloat);
+void (*GLFunctions::glUniform1fv)(GLint, GLsizei, const GLfloat *);
+void (*GLFunctions::glUniform1i)(GLint, GLint);
+void (*GLFunctions::glUniform1iv)(GLint, GLsizei, const GLint *);
+void (*GLFunctions::glUniform2f)(GLint, GLfloat, GLfloat);
+void (*GLFunctions::glUniform2fv)(GLint, GLsizei, const GLfloat *);
+void (*GLFunctions::glUniform2i)(GLint, GLint, GLint);
+void (*GLFunctions::glUniform2iv)(GLint, GLsizei, const GLint *);
+void (*GLFunctions::glUniform3f)(GLint, GLfloat, GLfloat, GLfloat);
+void (*GLFunctions::glUniform3fv)(GLint, GLsizei, const GLfloat *);
+void (*GLFunctions::glUniform3i)(GLint, GLint, GLint, GLint);
+void (*GLFunctions::glUniform3iv)(GLint, GLsizei, const GLint *);
+void (*GLFunctions::glUniform4f)(GLint, GLfloat, GLfloat, GLfloat, GLfloat);
+void (*GLFunctions::glUniform4fv)(GLint, GLsizei, const GLfloat *);
+void (*GLFunctions::glUniform4i)(GLint, GLint, GLint, GLint, GLint);
+void (*GLFunctions::glUniform4iv)(GLint, GLsizei, const GLint *);
+void (*GLFunctions::glUniformMatrix2fv)(GLint, GLsizei, GLboolean, const GLfloat *);
+void (*GLFunctions::glUniformMatrix3fv)(GLint, GLsizei, GLboolean, const GLfloat *);
+void (*GLFunctions::glUniformMatrix4fv)(GLint, GLsizei, GLboolean, const GLfloat *);
+void (*GLFunctions::glUseProgram)(GLuint);
+void (*GLFunctions::glValidateProgram)(GLuint);
+void (*GLFunctions::glVertexAttrib1f)(GLuint, GLfloat);
+void (*GLFunctions::glVertexAttrib1fv)(GLuint, const GLfloat *);
+void (*GLFunctions::glVertexAttrib2f)(GLuint, GLfloat, GLfloat);
+void (*GLFunctions::glVertexAttrib2fv)(GLuint, const GLfloat *);
+void (*GLFunctions::glVertexAttrib3f)(GLuint, GLfloat, GLfloat, GLfloat);
+void (*GLFunctions::glVertexAttrib3fv)(GLuint, const GLfloat *);
+void (*GLFunctions::glVertexAttrib4f)(GLuint, GLfloat, GLfloat, GLfloat, GLfloat);
+void (*GLFunctions::glVertexAttrib4fv)(GLuint, const GLfloat *);
+void (*GLFunctions::glVertexAttribPointer)(GLuint, GLint, GLenum, GLboolean, GLsizei, const void *);
+void (*GLFunctions::glViewport)(GLint, GLint, GLsizei, GLsizei);
+
+#ifndef MBGL_USE_GLES2
+void (*GLFunctions::glDrawPixels)(GLsizei, GLsizei, GLenum, GLenum, const GLvoid *);
+void (*GLFunctions::glGetDoublev)(GLenum, GLdouble *);
+void (*GLFunctions::glPixelTransferf)(GLenum, GLfloat);
+void (*GLFunctions::glPixelZoom)(GLfloat, GLfloat);
+void (*GLFunctions::glPointSize)(GLfloat);
+void (*GLFunctions::glRasterPos4d)(GLdouble, GLdouble, GLdouble, GLdouble);
+#endif
+
+void GLFunctions::init() {
```

Comment:
Can we do this in a way where this function is implemented directly by the various platforms? This means we don't have to copy around all of the function pointers and will hopefully end up with a smaller binary size increase.

Reply:
I think you are right, we don't need `GLFunctions` and `GLFunctionsPointers`.